src/private/warehouse.ts

```diff
@@ -15,6 +16,65 @@ import {
     warehouseContract
 } from "../services/warehouse";
 
+const WarehouseItemSchema = yup.object({
+    available_quantity: yup
+        .number()
+        .typeError("available_quantity must be a number.")
+        .required("The available_quantity (integer) must be provided."),
+    no_update_after: yup
+        .string()
+        .strict()
+        .typeError("available_quantity must be a number.")
+        .optional(),
+    item_id: yup
+        .number()
+        .typeError("item_id must be a number.")
+        .positive("item_id must be a positive number.")
+        .required("The item_id (integer) must be provided."),
+    name: yup
+        .string()
+        .strict()
+        .typeError("name must be a string.")
+        .required("The name (string) must be provided."),
+    data: yup.lazy((value) => {
```

Comment:
likewise, this would just pass the `dataValidator` reference to the `data:` field